### PR TITLE
Documentation updated: Hosting from a sub path section added

### PR DESCRIPTION
**Pull request by ummarbhutta**
Documentation updated regarding hosting on a subpath when using reverse proxy.

### CODE REVIEW

**Comment by stianst** [comment]
It is not needed to change Keycloak's context path when you have a different path on the proxy. Instead use hostname-path or hostname-url.

**Comment by ummarbhutta** [comment]
Hi @stianst , I have not tried those configuration environment variables,  `KC_HTTP_RELATIVE_PATH` seems most appropriate to me for this configuration and worked for me. 

Reason for this PR is that this configuration should be documented with example on this readme file for future reference.

**Comment by stianst** [comment]
@ummarbhutta As I said before you don't need to change Keycloak's context-path to expose it on a different context-path on the reverse proxy, which is what you're doing in the example here. Just use config options I mentioned to configure where Keycloak is exposed on the proxy.

**Comment by ummarbhutta** [comment]
Thanks @stianst , Agree with that.

Q: Do you agree that this (the strategy you are referring to) needs to be documented on reverse proxy guide page?

File: docs/guides/src/main/server/reverseproxy.adoc [hunk 1]
```diff
@@ -43,6 +43,10 @@ Take extra precautions to ensure that the X-Forwarded-For header is set by your
 If this header is incorrectly configured, rogue clients can set this header and trick Keycloak into thinking the client is connected from a different IP address than the actual address.
 This precaution can be more critical if you do any deny or allow listing of IP addresses.
 
+== Hosting keyclock on subpath
```
Comment (stianst) [inline docs/guides/src/main/server/reverseproxy.adoc:46]:
```suggestion
== Different context-path on reverse proxy
```

File: docs/guides/src/main/server/reverseproxy.adoc [hunk 2]
```diff
@@ -43,6 +43,10 @@ Take extra precautions to ensure that the X-Forwarded-For header is set by your
...
+
+When Keycloak is hosted behind a reverse proxy, it is usually served from a subpath (not from rool url), for this to work properly Keyclock should be configured to serve resources from a subpath. Use `KC_HTTP_RELATIVE_PATH` for this, e.g. if you want to set Keyclock to host from subpath `/auth` then set this environment variable to `/auth`, see details https://www.keycloak.org/server/all-config?f=build
```
Comment (stianst) [inline docs/guides/src/main/server/reverseproxy.adoc:48]:
```suggestion
Keycloak assumes it is exposed through the reverse proxy under the same context path as Keycloak is configured for. By default Keycloak is exposed through the root (`/`).

You can use `hostname-path` or `hostname-url` in these cases, or alternatively you can also change the context path of Keycloak itself to match the context path for the reverse proxy using the `http-relative-path` option.

When Keycloak is hosted behind a reverse proxy, it is usually served from a subpath (not from rool url), for this to work properly Keyclock should be configured to serve resources from a subpath. Use `KC_HTTP_RELATIVE_PATH` for this, e.g. if you want to set Keyclock to host from subpath `/auth` then set this environment variable to `/auth`, see details https://www.keycloak.org/server/all-config?f=build
```

**Comment by stianst** [comment]
> Thanks @stianst , Agree with that.
> 
> Q: Do you agree that this (the strategy you are referring to) needs to be documented on reverse proxy guide page?

Yeah, I agree this will be good to cover in the guide. I've made some suggestions to your PR.

**Comment by ummarbhutta** [comment]
@stianst , Updated PR as per comments.

File: docs/guides/src/main/server/reverseproxy.adoc [hunk 3]
```diff
@@ -43,6 +43,11 @@ Take extra precautions to ensure that the X-Forwarded-For header is set by your
 If this header is incorrectly configured, rogue clients can set this header and trick Keycloak into thinking the client is connected from a different IP address than the actual address.
 This precaution can be more critical if you do any deny or allow listing of IP addresses.
 
+== Different context-path on reverse proxy
+
+Keycloak assumes it is exposed through the reverse proxy under the same context path as Keycloak is configured for. By default Keycloak is exposed through the root (`/`).
+You can use `hostname-path` or `hostname-url` in these cases, or alternatively you can also change the context path of Keycloak itself to match the context path for the reverse proxy using the `http-relative-path` option. e.g. to configure it using `http-relative-path`, if you want to set Keyclock to host from subpath `/auth` then set `http-relative-path` to `/auth`, see details https://www.keycloak.org/server/all-config?f=build
```
Comment (stianst) [inline docs/guides/src/main/server/reverseproxy.adoc:49]:
```suggestion
Keycloak assumes it is exposed through the reverse proxy under the same context path as Keycloak is configured for. By default Keycloak is exposed through the root (`/`), which means it expects to be exposed through the reverse proxy on `/` as well.

You can use `hostname-path` or `hostname-url` in these cases, for example using `--hostname-path=/auth` if Keycloak is exposed through the reverse proxy on `/auth`.
 
Alternatively you can also change the context path of Keycloak itself to match the context path for the reverse proxy using the `http-relative-path` option, which will change the context-path of Keycloak itself to match the context path used by the reverse proxy.
```

**Comment by stianst** [comment]
Reading it again, maybe it could be improved a little. Added a suggestion.

**Comment by ummarbhutta** [comment]
@stianst looks good, have updated it.

File: docs/guides/src/main/server/reverseproxy.adoc [hunk 4]
```diff
@@ -43,6 +43,13 @@ Take extra precautions to ensure that the X-Forwarded-For header is set by your
 If this header is incorrectly configured, rogue clients can set this header and trick Keycloak into thinking the client is connected from a different IP address than the actual address.
 This precaution can be more critical if you do any deny or allow listing of IP addresses.
 
+== Different context-path on reverse proxy
+
+Keycloak assumes it is exposed through the reverse proxy under the same context path as Keycloak is configured for. By default Keycloak is exposed through the root (`/`), which means it expects to be exposed through the reverse proxy on `/` as well.
+You can use `hostname-path` or `hostname-url` in these cases, for example using `--hostname-path=/auth` if Keycloak is exposed through the reverse proxy on `/auth`.
+ 
+Alternatively you can also change the context path of Keycloak itself to match the context path for the reverse proxy using the `http-relative-path` option, which will change the context-path of Keycloak itself to match the context path used by the reverse proxy. see details of configuration options at https://www.keycloak.org/server/all-config?f=build
```
[the remark this answers is not in the document]
Comment (ummarbhutta) [reply docs/guides/src/main/server/reverseproxy.adoc]: Done, wasn't aware of `includedOption` syntax, but that is nice feature, I hope I did it in right way.

**Review by stianst — APPROVED** [review]

**Pull request merged**